extensions/android-paging3/src/main/java/app/cash/sqldelight/paging3/QueryPagingSource.kt

```diff
@@ -70,6 +71,35 @@ fun <RowType : Any> QueryPagingSource(
   context,
 )
 
+/**
+ * Variant of [QueryPagingSource] that accepts a [Long] instead of an [Int] for [countQuery]
+ * and [queryProvider].
+ *
+ * If the result of [countQuery] exceeds [Int.MAX_VALUE], then the count will be truncated
+ * to the least significant 32 bits of this [Long] value.
+ *
+ * @see toInt
+ */
+@Suppress("FunctionName")
+fun <RowType : Any> QueryPagingSourceLong(
```

Comment:
I'd prefer having this be the same name as the other function (`QueryPagingSource`) but we get a `Platform declaration clash` when doing so. Open to better names

Reply:
Should we just switch them so the Int one has a fancy name since SQLDelight users will pretty much only use this one?

Reply:
Using `@JvmName` should let you keep the names the same in Kotlin

Reply:
Nice, just renamed it to `QueryPagingSource` after specifying `@JvmName`.

@AlecStrong I just suffixed the type to each one for consistency in the `@JvmName`. I'm assuming that no one's using SQLDelight with Java.

Reply:
yep, perfect